### PR TITLE
[MO] turn on bom_tests

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
BOM tests were skipped because of changed folder structure. Enabled them back and corrected fixture for mo py files prefixes and mo_root directory setting so that list of files is correctly loaded from package_BOM and correctly compared with existing files in 'openvino_project_root/tools/mo/openvino/tools'.

Also removed `install_prerequisites` since they are obsolete and even cause problems when bom_test is run in CI they are not copied into package therefore removed them from the repository as well.

Also corrected messages accordingly to install from `pip install openvino` instead of `install_prerequisites`. 

### Tickets:
 - xxx-83790

### CODE REVIEW

**Comment by mvafin** [comment]
Looks like BOM tests are failing in CI

**Comment by pavel-esir** [comment]
> Looks like BOM tests are failing in CI

yes, i am currently solving it.

**Comment by pavel-esir** [comment]
> Looks like BOM tests are failing in CI

done. @mvafin please take a look

File: tools/mo/setup.py [hunk 1]
```diff
@@ -102,6 +102,7 @@ def find_package_modules(self, package, package_dir):
       'openvino.tools.mo.front.mxnet': ['*.json'],
       'openvino.tools.mo.front.onnx': ['*.json'],
       'openvino.tools.mo.front.tf': ['*.json'],
+      'openvino.tools.mo.front.caffe': ['CustomLayersMapping.xml*']
```
Comment (mvafin) [inline tools/mo/setup.py:105]: Why this change?

File: tools/mo/unit_tests/mo/bom_test.py [hunk 2]
```diff
@@ -7,16 +7,19 @@
...
 
+from openvino.tools.mo.utils.utils import get_mo_root_dir
+
 dir_patterns_to_skip = ['.*__pycache__.*']
 file_patterns_to_skip = ['.*\\.DS_Store$',
                          '.*\\.swp',
-                         '.*\\.pyc$']
+                         '.*\\.pyc$',
+                         'requirements.*\.txt',
+                         '.*version.txt']
```
Comment (mvafin) [inline tools/mo/unit_tests/mo/bom_test.py:17]: I guess it can be just `'version.txt'`

File: tools/mo/automation/package_BOM.txt [hunk 3]
```diff
@@ -1104,11 +1096,4 @@ openvino/tools/mo/utils/type_utils.py
 openvino/tools/mo/utils/unsupported_ops.py
 openvino/tools/mo/utils/utils.py
 openvino/tools/mo/utils/version.py
-openvino/tools/mo/utils/versions_checker.py
-requirements.txt
```
Comment (mvafin) [inline tools/mo/automation/package_BOM.txt:1108]: Why we do not check requirements files?

File: tools/mo/unit_tests/mo/bom_test.py [hunk 4]
```diff
@@ -68,36 +70,32 @@ def test_bom_file(self):
                 print(f.replace('\\', '/'))
         self.assertTrue(not len(missing_files), '{} files missed in BOM'.format(len(missing_files)))
 
-    @unittest.skip("83076 - test infra change affects bom tests")
     def test_bom_does_not_contain_unittest_files(self):
         for file_name in self.existing_files:
             self.assertFalse(file_name.endswith('_test.py'), 'BOM file contains test file {}'.format(file_name))
 
-    @unittest.skip("83076 - test infra change affects bom tests")
+    # @unittest.skip("83076 - test infra change affects bom tests")
```
Comment (mvafin) [inline tools/mo/unit_tests/mo/bom_test.py:77]: Remove?

Comment (pavel-esir) [reply tools/mo/unit_tests/mo/bom_test.py:77]: thanks for noticing. will remove that

File: tools/mo/setup.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/setup.py:105]: This file is needed by Caffe models and by mistake was missed in when mo is installed through pip. It should be copied to wheel package.
And because this file skipping during check we didn't get any bom error even when they are turned on.

File: tools/mo/unit_tests/mo/bom_test.py (hunk 2 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/bom_test.py:17]: yeah, `version.txt` also should be enough. 
But can we fix that separately? With bunch of other changes it they will be requested

File: tools/mo/unit_tests/mo/bom_test.py (hunk 4 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/bom_test.py:77]: same as previous. i think it's `nit` and can be fixed with bunch of other changes if they will be needed.

**Comment by pavel-esir** [comment]
@mvafin can you please approve? I think left comment can be considered as `nit`s and in order not to run overload CI i will fix them in a bunch of other changes. 
We have a green CI which was not easy to achieve :)

File: tools/mo/unit_tests/mo/bom_test.py (hunk 2 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/bom_test.py]: done

File: tools/mo/unit_tests/mo/bom_test.py (hunk 4 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/bom_test.py]: done

**Review by mvafin — APPROVED** [review]

File: tools/mo/setup.py (hunk 1 above)
Comment (rkazants) [reply tools/mo/setup.py:105]: This file is mentioned nowhere now. How is it used and for what are not clear? I did not find it in the documentation. Please add this in the documentation where we have an example with custom mapping. Let users see the example of such mapping.

File: tools/mo/setup.py [hunk 5]
```diff
@@ -31,7 +31,7 @@ def read_text(path):
 requirements_txt = []
 py_modules = []
 for item in os.listdir():
-    if re.match(r'requirements(.*)\.txt', item):
+    if re.match(r'requirements_?(tf|tf2|onnx|mxnet|kaldi|caffe)?\.txt', item):
```
Comment (rkazants) [inline tools/mo/setup.py:34]: I disagree with this change because we should always update it once there are new requirement files, as example,  for pytorch and paddle. Let us not change it.

**Review by rkazants — COMMENTED** [review]
Left two comments. Please address, The rest looks good to me.

File: tools/mo/setup.py (hunk 5 above)
Comment (pavel-esir) [reply tools/mo/setup.py:34]: I understand what you mean and agree that probably it could be easier with `(.*)` but anyways if we add new requirements we still will need to add them separately below in the same `setup.py` 

https://github.com/openvinotoolkit/openvino/blob/4dd75a1b8be4351698a5f8a488a53eacd85a4ba4/tools/mo/setup.py#L107-L112

And one more factor. The reason i specified them manually is that `(.*)` copied also `requirements_dev.txt` which copies file with packages (pytest, generate and so on) that are not necessary when openvino-dev is installed from pip/wheel. Copying `requirements-dev.txt` didn't cause significant damage but still was not quite right previously.

Therefore i insist to keep this regexp pattern with manual setting of tf/tf_2/onnx/and so on

Comment (rkazants) [reply tools/mo/setup.py:34]: Will it work for `requirement.txt` case?

Please fix the comment above about `CustomLayersMapping.xml*` because now we are adding new file into the package but it is not clear for what.

File: tools/mo/setup.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/setup.py:105]: There are some documentation about CustomLayersMapping for Caffe. Exactly this file `CustomLayersMapping.xml.example` is not mentioned 
but `CustomLayersMapping.xml` is mentioned here https://github.com/openvinotoolkit/openvino/blob/master/docs/MO_DG/prepare_model/convert_model/Convert_Model_From_Caffe.md

I didn't add this new file I just returned it back when it was at the package, e.g. at 2021.4 it was copied to model_optimizer during installation 
image
but when building package switched to pip/wheels we lost it. Here I returned it back.

Comment (rkazants) [reply tools/mo/setup.py:105]: Now this is useless file in our package. Let us make some value for this file by adding just one sentence about it. Believe - not a big effort:)

Comment (pavel-esir) [reply tools/mo/setup.py:105]: I agree it's not a big effort :) 
I am afraid of the CI validation problems. I will add such line into documentation and try to push, maybe CI will see that only documentation was changed and will not full e2e validation. Let's see if this works :)

Comment (rkazants) [reply tools/mo/setup.py:105]: CI is not a problem. During the weekend I obtained three PRs with green CI.

File: tools/mo/setup.py (hunk 5 above)
Comment (pavel-esir) [reply tools/mo/setup.py:34]: Indeed `requirements.txt` is also copied by this regex. And it's right to be copied because it's used subsequently 
https://github.com/openvinotoolkit/openvino/blob/940844e71f6fa5603576b2d0d51da043a5634b0a/tools/mo/openvino/tools/mo/utils/versions_checker.py#L247-L258

File: tools/mo/setup.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/setup.py:105]: Mentioned purpose of `CustomLayersMapping.xml.example` in the docs

**Review by rkazants — APPROVED** [review]

**Comment by pavel-esir** [comment]
@rkazants addressed comments. Please approve
